src/workarounds/forwarddiff_rules.jl

```diff
@@ -272,3 +272,32 @@ function Smearing.occupation(S::Smearing.FermiDirac, d::ForwardDiff.Dual{T}) whe
     end
     ForwardDiff.Dual{T}(Smearing.occupation(S, x), ∂occ * ForwardDiff.partials(d))
 end
+
+# Workarounds for issue https://github.com/JuliaDiff/ForwardDiff.jl/issues/324
+ForwardDiff.derivative(f, x::Complex) = throw(DimensionMismatch("derivative(f, x) expects that x is a real number (does not support Wirtinger derivatives). Separate real and imaginary parts of the input."))
```

Comment:
These fixes are not for the issue above. Is the fix not already upstream in a released version? If yes, only define the functions if the version is below the one that has the fix (and then at some point we remove the code and depend on a new version)

Reply:
This https://github.com/JuliaDiff/ForwardDiff.jl/pull/577 seems to fix the exponentiation problem. So maybe remove it altogether?

Reply:
Does it fix https://github.com/JuliaDiff/ForwardDiff.jl/issues/514#issuecomment-1113274482? Since my message was after that PR was merge I don't think so?

Reply:
Right, I thought is was after. This is strange, something seems to have fixed the issue, because I am not running into it anymore, even for this test:
```
using ForwardDiff, FiniteDifferences, Random, Test

v = randn()
p, m = randn(ComplexF64), randn(ComplexF64)

for f in (x -> (x*m)^p,
          x -> m^(p*x),
          x -> (x*m)^(p*x))
  @test ≈(ForwardDiff.derivative(f, v), central_fdm(5,1)(f, v), atol=1e-8)
end
```

Reply:
What about the code in my comment in the forwarddiff tracker above?